server/services/rules_engine.py

```
@@ -213,6 +215,65 @@ def score_fit_match(items: list[dict], body_measurements: dict | None) -> float:
+    if angle > 180:
+        angle = 360 - angle
+    return max(0.0, 1.0 - angle / 180.0)
+
+
+def _best_palette_similarity(hex_color: str, palette: list[str]) -> float | None:
+    scores = []
+    for palette_hex in palette:
+        try:
+            scores.append(_hue_similarity(hex_color, palette_hex))
+        except ValueError:
+            continue
```

Ruff (0.15.17)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)